src/solver/expressions/visitors/CloneVisitor.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#include <antares/solver/expressions/nodes/ExpressionsNodes.h>
+#include <antares/solver/expressions/visitors/CloneVisitor.h>
+
+namespace Antares::Solver::Visitors
+{
+CloneVisitor::CloneVisitor(Registry<Nodes::Node>& mem):
+    registry_(mem)
+{
+}
+
+Nodes::Node* CloneVisitor::visit(const Nodes::AddNode& input_node)
+{
+    return registry_.create<Nodes::AddNode>(dispatch(*input_node.left()),
+                                            dispatch(*input_node.right()));
+}
+
+Nodes::Node* CloneVisitor::visit(const Nodes::SubtractionNode& input_node)
```

Comment:
Same as above  : input_node ==> **SubstractionNode**, and if we feel it's too long, **SubstNode**. But why not the first name (which is the clearest) ?

`LessThanOrEqualNode& input_node` ==> `LessThanOrEqualNode& lessNode` ?
`VariableNode& param` ==> `VariableNode& varNode`

Reply:
Or just `node`, see comment above.